csharp-intermediate-topics/LockingMechanismInCSharp/LockingMechanismInCSharp/DeadLock.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace LockingMechanismInCSharp
+{
+    public static class DeadLock
+    {
+        private static int _counter;
+        private static readonly object _firstLockObject = new();
+        private static readonly object _secondLockObject = new();
+
+        public static int Execute()
+        {
+            var thread1 = new Thread(IncrementCounter1);
+            var thread2 = new Thread(IncrementCounter2);
+
+            thread1.Start();
+            thread2.Start();
+
+            thread1.Join();
+            thread2.Join();
+
+            return _counter;
+        }
+
+        private static void IncrementCounter1()
+        {
+            lock (_firstLockObject)
+            {
+                Thread.Sleep(1000);
+
+                lock (_secondLockObject)
+                {
+                    for (int i = 1; i <= 100000; i++)
```

Comment:
This comment applies across all of the for loops in the code. The more standard approach for a loop that is counting is to start from 0 and check `<` as opposed to starting at 1 and performing `<=`. There are cases where you may need to start at 1 or some other index, but in the standard "I want to do something 100 times", industry practice is to start at 0. Since that is a standard, it also makes it a half step quicker for someone stepping into the codebase to parse what is going on with the loop. Seeing it start at 1 gives an immediate pause to try to think why it isn't starting at 0.

Reply:
fixed, also changed int to var